detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/CanBeNonNullable.kt

```diff
@@ -231,13 +233,22 @@ class CanBeNonNullable(config: Config = Config.empty) : Rule(config) {
         }
 
         override fun visitWhenExpression(expression: KtWhenExpression) {
-            val subjectDescriptor = expression.subjectExpression
-                ?.let { it as? KtNameReferenceExpression }
-                ?.getResolvedCall(bindingContext)
-                ?.resultingDescriptor
+            val nullCheckedDescriptor =
+                expression.subjectExpression?.collectDescendantsOfType<KtNameReferenceExpression>().orEmpty()
+                    .mapNotNull {
+                        val callDescriptor =
+                            it.getResolvedCall(bindingContext)
+                        val valueParameterDescriptor =
+                            callDescriptor?.resultingDescriptor as? ValueParameterDescriptor
+                        if (valueParameterDescriptor != null && callDescriptor.getReturnType().isNullable()) {
+                            valueParameterDescriptor
+                        } else {
+                            null
+                        }
+                    }
```

Comment:
```suggestion
            val nullCheckedDescriptor = expression.subjectExpression?
                .collectDescendantsOfType<KtNameReferenceExpression>()
                .orEmpty()
                .mapNotNull { it.getResolvedCall(bindingContext) }
                .filter { callDescriptor -> callDescriptor.getReturnType().isNullable() }
                .mapNotNull { callDescriptor -> callDescriptor.resultingDescriptor as? ValueParameterDescriptor }
```

What do you think? Is it easier to read?